src/Course/Optional.hs

```diff
@@ -61,7 +61,10 @@ contains a1 (Full a2) = a1 == a2
 --     a) What are the data constructors of `Maybe`? What are their types?
 --     b) What instances does `Maybe` derive? In your answer, state the instance context and the instance head.
 question_Optional_1 :: (String, String)
-question_Optional_1 = error "todo"
+question_Optional_1 =
+  ( "The data constructors of maybe are `Just a` or `Nothing` which are both of type `Maybe a`."
```

Comment:
> which are both of type `Maybe a`

Let's be careful here. `Nothing` certainly has type `Maybe a`. `Just` is the data constructor, not `Just a`. `Just a` is syntax to show us that `Just` carries an `a` value in it, so the type of `Just` is `a -> Maybe a`.